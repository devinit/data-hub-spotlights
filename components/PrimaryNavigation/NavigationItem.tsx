import React, { FunctionComponent } from 'react';
import classNames from 'classnames';

export interface NavigationItemProps {
  title: string;
  active: boolean;
  url: string;
}

const NavigationItem: FunctionComponent<NavigationItemProps> = ({ active, title, url }) => {
  return (
    <li className={classNames('navigation-primary__item', { 'navigation-primary__item--active': active })}>
      <a href={url}>{title}</a>
    </li>
  );
};

export { NavigationItem };
