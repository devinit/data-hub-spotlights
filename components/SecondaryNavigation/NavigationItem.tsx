import React, { FunctionComponent } from 'react';
import classNames from 'classnames';

export interface NavigationItemProps {
  title: string;
  active: boolean;
  url: string;
}

const NavigationItem: FunctionComponent<NavigationItemProps> = ({ active, title, url }) => {
  return (
    <li className={classNames('navigation-secondary__item', { 'navigation-secondary__item--active': active })}>
      <a href={url}>{title}</a>
    </li>
  );
};

export { NavigationItem };
