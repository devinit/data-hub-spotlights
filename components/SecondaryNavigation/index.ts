export * from './NavigationItem';
export * from './SecondaryNavigation';
