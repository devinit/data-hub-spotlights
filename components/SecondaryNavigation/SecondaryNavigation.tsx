import classNames from 'classnames';
import React, { FunctionComponent } from 'react';
import { NavigationItem, NavigationItemProps } from './NavigationItem';

interface SecondaryNavigationProps {
  items?: NavigationItemProps[];
  className?: string;
}

const SecondaryNavigation: FunctionComponent<SecondaryNavigationProps> = ({ children, className, items }) => (
  <nav className={classNames('navigation-secondary', className)}>
    <ul className="navigation-secondary__items" data-testid="navigation-secondary-ul">
      {items?.map(item => (
        <NavigationItem key={item.title} {...item} />
      ))}
      {children}
    </ul>
  </nav>
);

SecondaryNavigation.defaultProps = { items: [] };

export { SecondaryNavigation };
