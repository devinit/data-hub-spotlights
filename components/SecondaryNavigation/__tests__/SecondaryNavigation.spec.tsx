/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { SecondaryNavigation } from '../SecondaryNavigation';
import { NavigationItemProps, NavigationItem } from '../NavigationItem';

const items: NavigationItemProps[] = [
  {
    title: 'Nav 1',
    url: 'https://localhost:3000/nav-1',
    active: false
  },
  {
    title: 'Nav 2',
    url: 'https://localhost:3000/nav-2',
    active: true
  }
];

describe('SecondaryNavigation', () => {
  test('renders correctly', () => {
    const renderer = TestRenderer.create(<SecondaryNavigation />).toJSON();

    expect(renderer).toMatchSnapshot();
  });

  test('renders with specified className', () => {
    const { container } = render(<SecondaryNavigation className="spotlight-secondary-nav" />);

    expect(container.firstChild).toHaveClass('spotlight-secondary-nav');
  });

  test('renders with specified navigation items', () => {
    const renderer = TestRenderer.create(<SecondaryNavigation items={items} />).toJSON();

    expect(renderer).toMatchSnapshot();
  });

  test('renders with specified children (preferably NavigationItem)', () => {
    const renderer = TestRenderer.create(
      <SecondaryNavigation>
        <NavigationItem title="My Nav" url="/my-nav" active />
      </SecondaryNavigation>
    ).toJSON();

    expect(renderer).toMatchSnapshot();
  });
});
