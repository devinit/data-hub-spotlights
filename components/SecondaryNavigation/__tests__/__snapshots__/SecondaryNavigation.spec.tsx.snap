// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecondaryNavigation renders correctly 1`] = `
<nav
  className="navigation-secondary"
>
  <ul
    className="navigation-secondary__items"
    data-testid="navigation-secondary-ul"
  />
</nav>
`;

exports[`SecondaryNavigation renders with specified children (preferably NavigationItem) 1`] = `
<nav
  className="navigation-secondary"
>
  <ul
    className="navigation-secondary__items"
    data-testid="navigation-secondary-ul"
  >
    <li
      className="navigation-secondary__item navigation-secondary__item--active"
    >
      <a
        href="/my-nav"
      >
        My Nav
      </a>
    </li>
  </ul>
</nav>
`;

exports[`SecondaryNavigation renders with specified navigation items 1`] = `
<nav
  className="navigation-secondary"
>
  <ul
    className="navigation-secondary__items"
    data-testid="navigation-secondary-ul"
  >
    <li
      className="navigation-secondary__item"
    >
      <a
        href="https://localhost:3000/nav-1"
      >
        Nav 1
      </a>
    </li>
    <li
      className="navigation-secondary__item navigation-secondary__item--active"
    >
      <a
        href="https://localhost:3000/nav-2"
      >
        Nav 2
      </a>
    </li>
  </ul>
</nav>
`;
